src/Tuura/Concept/Circuit/Basic.hs

```diff
@@ -35,17 +36,19 @@ data Concept s e a = Concept
                    {
                        initial   :: a -> InitialValue,
                        arcs      :: [([e], e)],
-                       interface :: a -> Interface
+                       interface :: a -> Interface,
+                       invariant :: [[e]]
                    }
```

Comment:
That would work better. I will start by doing it for Invariant, then for Causality later.